### PR TITLE
Allow parametric noise overrides in noise formula for residuals

### PR DESCRIPTION
**Pull request by yannikschaelte**

### CODE REVIEW

**Comment by yannikschaelte** [comment]
@dweindl does this capture all replacement variants we allow?

File: petab/calculate.py [hunk 1]
```diff
@@ -95,16 +105,81 @@ def calculate_residuals_for_table(
...
+
+    Returns:
+        noise_formulas: Dictionary of {observable_id}: {noise_formula}.
+    """
+    noise_formulas = {}
+    # iterate over observables
+    for row in observable_df.itertuples():
+        observable_id = row.Index
+        if NOISE_FORMULA not in observable_df.columns:
+            noise_formula = []
```
Comment (dweindl) [inline petab/calculate.py]: Is empty list preferable over None?

File: petab/calculate.py [hunk 2]
```diff
@@ -12,6 +14,7 @@ def calculate_residuals(
         measurement_dfs: Union[List[pd.DataFrame], pd.DataFrame],
         simulation_dfs: Union[List[pd.DataFrame], pd.DataFrame],
         observable_dfs: Union[List[pd.DataFrame], pd.DataFrame],
+        parameter_df: Union[List[pd.DataFrame], pd.DataFrame],
```
Comment (dweindl) [inline petab/calculate.py]:
```suggestion
        parameter_dfs: Union[List[pd.DataFrame], pd.DataFrame],
```
Consistency with other arguments?

Comment (yannikschaelte) [reply petab/calculate.py]: ah right. that would depend also on if we now allow an array of parameter dfs in yaml.

File: petab/calculate.py (hunk 1 above)
Comment (yannikschaelte) [reply petab/calculate.py]: no

**Pull request merged**